sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -0,0 +1,359 @@
+        suffix_table = { 'euvi':'n7eu{char}.fts', 'cor2': 'd7c2{char}.fts', 'hi1':'s7h1{char}.fts', 'hi2':'s7h2{char}.fts'}
+        url_pattern = prefix + suffix_table[detector]
+        crawler = Scraper(url_pattern, source=table_[source], det=table[detector], char=table_[source][0].upper())
+        result = crawler.filelist(timerange)
+        return result
+
+    def _makeimap(self):
+        """
+        Helper Function: used to hold information about source.
+        """
+        self.map_['instrument'] = 'secchi'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Answers whether client can service the query.
+
+        Parameters
+        ----------
+        query : list of query objects
+
+        Returns
+        -------
+        boolean: answer as to whether client can service the query
+
+        """
+        regex = re.compile('[^a-zA-Z0-9]')
+        chkattr = ['Time', 'Instrument', 'Source', 'Detector']
+        chklist = [x.__class__.__name__ in chkattr for x in query]
+        chk_var = 0
+        sources = ['STEREO_A', 'STEREO_B']
+        detectors = ['euvi', 'cor2', 'hi1', 'hi2']
+        for x in query:
+            if x.__class__.__name__ == 'Instrument' and x.value.lower() == 'secchi':
+                chk_var += 1
+            if x.__class__.__name__ == 'Source' and x.value in sources:
+                chk_var += 1
+            if x.__class__.__name__ == 'Detector' and regex.sub('', x.value).lower() in detectors:
+                chk_var += 1
+        return chk_var == 3
+
+
+class PLASTICClient(GenericClient):
+    """
+    Returns a list of URLS to PLASTIC files corresponding to value of input timerange.
+    URL source: `http://stereo-ssc.nascom.nasa.gov/data/beacon/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Fixed argument, case-insensitive = 'plastic'
+
+    Source: Two arguments, case-sensitive - 'STEREO_A' or 'STEREO_B'.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(a.Time('2007/3/2 17:00:00', '2007/3/4 17:15:00'),
+        a.Instrument('plastic'), a.Source('STEREO_A'))
+    >>> print(results)
+       [<Table length=2>
+         Start Time           End Time      Source Instrument
+           str19               str19         str5     str7
+        ------------------- ------------------- ------ ----------
+        2007-03-02 00:00:00 2007-03-03 00:00:00  ahead    plastic
+        2007-03-03 00:00:00 2007-03-04 00:00:00  ahead    plastic]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        if not timerange:
+            return []
+        source = kwargs.get('source')
```

Comment:
I disagree a bit, All that the ```Fido``` clients have in common is, that they accept a time range argument. Different clients accept different "compulsory" arguments. For e.g ```BBSO``` would take  ```Level```, ```GOES``` would take ```Physobs```, ```Kanzelhohe``` would take ```Wavelength``` so on, so forth.
If you want specific data from a client (If at all it provides them) then you have to mention that extra argument, so even though that extra argument is compulsory, when you take into account all Fido Clients (and we want all Fido clients to behave in some semblance of similar behavior, that is why they're Fido clients, they behave in the way we define them to), that argument is actually extra (**not compulsory**). Hence I put them in ```kwargs```.